ios/sdk/src/dropbox/Dropbox.m

```diff
@@ -128,25 +131,31 @@ + (BOOL)application:(UIApplication *)app
     if (currentReject == nil || currentResolve == nil) {
         return NO;
     }
-    DBOAuthResult *authResult = [DBClientsManager handleRedirectURL:url];
-    if (authResult) {
-        if ([authResult isSuccess]) {
-            currentResolve(authResult.accessToken.accessToken);
-        } else {
-            NSString *msg;
-            if ([authResult isError]) {
-                msg = [NSString stringWithFormat:@"%@, error type: %zd",[authResult errorDescription], [authResult errorType]];
+    
+    BOOL canHandle = [DBClientsManager handleRedirectURL:url completion:^(DBOAuthResult *authResult) {
+        if (authResult) {
+            if ([authResult isSuccess]) {
+                NSInteger msTimestamp = authResult.accessToken.tokenExpirationTimestamp * 1000;
+                NSDictionary *authInfo = @{@"token": authResult.accessToken.accessToken,
+                                           @"rToken": authResult.accessToken.refreshToken,
+                                           @"expireDate": @(msTimestamp)
+                };
+                currentResolve(authInfo);
             } else {
-                msg = @"OAuth canceled!";
+                NSString *msg;
+                if ([authResult isError]) {
+                    msg = [NSString stringWithFormat:@"%@, error type: %zd", [authResult errorDescription], [authResult errorType]];
+                } else {
```

Comment:
How is it possible to get here? Both `authResult isSuccess` and `uthResult isError` would need to return false, which feels weird.

Reply:
Again, this is how the previous implementation was done. This sample code is from the dropbox documentation. Should answer your question:

```
if (authResult != nil) {
      if ([authResult isSuccess]) {
        NSLog(@"\n\nSuccess! User is logged into Dropbox.\n\n");
      } else if ([authResult isCancel]) {
        NSLog(@"\n\nAuthorization flow was manually canceled by user!\n\n");
      } else if ([authResult isError]) {
        NSLog(@"\n\nError: %@\n\n", authResult);
      }
```

Reply:
Ah, there is an isCancel, ok!